src/components/ScrollTo.js

```diff
@@ -1,117 +1,57 @@
-import { cloneElement, createRef, Component } from 'react';
+import { cloneElement, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import isEmpty from 'lodash/isEmpty';
+import ns from '../config/css-ns';
+
+/**  */
+function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+  return ref.current;
+}
 
 /**
  * ScrollTo ~
 */
-export class ScrollTo extends Component {
-  /** */
-  constructor(props) {
-    super(props);
-
-    this.scrollToRef = createRef();
-  }
-
-  /** */
-  componentDidMount() {
-    const { scrollTo } = this.props;
-    if (!scrollTo) return;
-
-    this.scrollToElement();
-  }
-
-  /**
-   * If the scrollTo prop is true (and has been updated) scroll to the selected element
-  */
-  componentDidUpdate(prevProps) {
-    const { scrollTo } = this.props;
-    if (scrollTo && (prevProps.scrollTo !== scrollTo)) {
-      this.scrollToElement();
-    }
-  }
-
-  /**
-   * Return the getBoundingClientRect() of the containerRef prop
-  */
-  containerBoundingRect() {
-    const { containerRef } = this.props;
+export function ScrollTo({
+  children, containerRef, offsetTop = 0, scrollTo, nodeId, ...otherProps
+}) {
+  const scrollToRef = useRef();
+  const prevScrollTo = usePrevious(scrollTo);
 
-    if (!containerRef || !containerRef.current) return {};
+  useEffect(() => {
+    if (!scrollTo || scrollTo === prevScrollTo) return;
```

Comment:
Could we add back some of the spirit of the deleted comment:
"If the scrollTo prop is true (and has been updated) scroll to the selected element"

It might just be me but I found it a little hard to first get that `prevScrollTo` and `scrollTo` were just the boolean prop and not any complex coordinates.